src/functionBased/components/TodoItem.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState, useEffect } from 'react';
+import { FaTrash } from 'react-icons/fa';
+import PropTypes from 'prop-types';
+import styles from './TodoItem.module.css';
+
+const TodoItem = ({
+  todo, handleChangeProps, deleteTodosProps, setUpdate,
+}) => {
+  const [editing, setEditing] = useState(false);
+
+  useEffect(() => () => 'Cleaning up...', []);
+
+  const handleEditing = () => {
+    setEditing(true);
+  };
+
+  const handleUpdatedDone = (event) => {
+    if (event.key === 'Enter') {
+      setEditing(false);
+    }
+  };
+
+  const completedStyle = {
+    fontStyle: 'italic',
+    color: '#595959',
+    opacity: 0.4,
+    textDecoration: 'line-through',
+  };
+
+  const { completed, id, title } = todo;
+  const viewMode = {};
+  const editMode = {};
+
+  if (editing) {
+    viewMode.display = 'none';
+  } else {
+    editMode.display = 'none';
+  }
+
+  return (
+    <li className={styles.item}>
+      <div onDoubleClick={handleEditing} style={viewMode}>
+        <input
+          type="checkbox"
+          checked={completed}
+          onChange={() => handleChangeProps(id)}
```

Comment:
Good job handling props with destructuring technique! but there is an issue with the checkbox onChange handler, once the checkbox's state is changed the entire screen goes blank. I have added a demo image to properly illustrate this behavior.

![demo](https://user-images.githubusercontent.com/82473943/154357165-53e96642-3445-43e5-a898-ed77cca62152.gif)